src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java

```diff
@@ -1450,8 +1450,25 @@ private void updateCaseBuilderForTransformCase(CaseData caseData, ResponseCaseDa
         builder
                 .ihtReferenceNumber(caseData.getIhtReferenceNumber())
                 .primaryApplicantAlias(caseData.getPrimaryApplicantAlias())
-                .solsExecutorAliasNames(caseData.getSolsExecutorAliasNames())
-                .solsDeceasedAliasNamesList(caseData.getSolsDeceasedAliasNamesList());
+                .solsExecutorAliasNames(caseData.getSolsExecutorAliasNames());
+
+        List<CollectionMember<AliasName>> deceasedAliasNames = EMPTY_LIST;
+        if (caseData.getDeceasedAliasNameList() != null) {
+            deceasedAliasNames = caseData.getDeceasedAliasNameList()
+                    .stream()
+                    .map(CollectionMember::getValue)
+                    .map(this::buildDeceasedAliasNameExecutor)
+                    .map(alias -> new CollectionMember<>(null, alias))
+                    .toList();
+        }
+        if (deceasedAliasNames.isEmpty()) {
+            builder
+                    .solsDeceasedAliasNamesList(caseData.getSolsDeceasedAliasNamesList());
+        } else {
+            builder
+                    .solsDeceasedAliasNamesList(deceasedAliasNames)
+                    .deceasedAliasNamesList(null);
+        }
```

Comment:
Is this what is needed? As in if there's values in the deceasedAliasNameList then solsDeceasedAliasNamesList is filled with deceasedAliasNameList. Wondering if there could be values in solsDeseasedAliasNamesList already?